pkg/apis/core/validation/validation.go

```diff
@@ -2286,6 +2291,43 @@ func validateDNSPolicy(dnsPolicy *core.DNSPolicy, fldPath *field.Path) field.Err
 	return allErrors
 }
 
+const (
+	maxDNSNameserver      = 3
+	maxDNSSearchPaths     = 6
+	maxDNSSearchListChars = 256
+)
+
+func validatePodDNSParams(dnsParams *core.PodDNSParams, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if dnsParams != nil {
+		if !utilfeature.DefaultFeatureGate.Enabled(features.CustomPodDNS) {
+			allErrs = append(allErrs, field.Forbidden(fldPath, "custom pod DNS is disabled by feature-gate"))
+		} else {
+			if len(dnsParams.Nameservers) > maxDNSNameserver {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("nameservers"), dnsParams.Nameservers, "must not have more than 3 nameservers"))
```

Comment:
fmt.Sprintf("must not have more than %v nameservers", maxDNSNameservers)

Reply:
Done.